app/policies/article_policy.rb

```diff
@@ -9,6 +9,6 @@ def edit?
     private
 
   def author?
-    @user.id == @record.user_id
+    @user&.id == @record.user_id
```

Comment:
Now, i think that this functionality need used in "edit?" method, because author method just check who is author of article: 
`def edit?
    author? if @user
  end
  alias destroy? edit?
  alias update? edit?

    private

  def author?
    @user.id == @record.user_id
  end`